berkeley-function-call-leaderboard/bfcl/model_handler/local_inference/phi.py

```diff
@@ -12,50 +11,45 @@ class PhiHandler(OSSHandler):
     def __init__(self, model_name, temperature) -> None:
         super().__init__(model_name, temperature)
 
+    @override
+    def decode_ast(self, result, language="Python"):
+        result = result.strip()
+        if result.startswith("```json"):
+            result = result[len("```json") :]
+        if result.startswith("```python"):
+            result = result[len("```python") :]
+        return super().decode_ast(result, language)
+
+    @override
+    def decode_execute(self, result):
+        funcs = re.findall(r"\[[^\[\]]+\]", result)
+        decoded_funcs = []
+        for func in funcs:
+            decode_output = ast_parse(func, language="Python")
+            decoded_funcs.extend(decode_output)
+
+        return convert_to_function_call(decoded_funcs)
+
     @override
     def _format_prompt(self, messages, function):
-        if "Phi-3-small" in self.model_name:
-            # Phi-3-small
+        formatted_prompt = ""
+
+        if "Phi-4-mini" in self.model_name:
+            # Phi-4-mini
             """
-            "bos_token": "<|endoftext|>",
-            "chat_template": "{{ bos_token }}{% for message in messages %}{{'<|' + message['role'] + '|>' + '\n' + message['content'] + '<|end|>\n' }}{% endfor %}{% if add_generation_prompt %}{{ '<|assistant|>\n' }}{% else %}{{ eos_token }}{% endif %}",
-            "eos_token": "<|endoftext|>",
+            "chat_template": "{% for message in messages %}{% if message['role'] == 'system' and 'tools' in message and message['tools'] is not none %}{{ '<|' + message['role'] + '|>' + message['content'] + '<|tool|>' + message['tools'] + '<|/tool|>' + '<|end|>' }}{% else %}{{ '<|' + message['role'] + '|>' + message['content'] + '<|end|>' }}{% endif %}{% endfor %}{% if add_generation_prompt %}{{ '<|assistant|>' }}{% else %}{{ eos_token }}{% endif %}"
             """
-            formatted_prompt = "<|endoftext|>"
-        else:
-            # Phi-3.5-mini, Phi-3-medium, Phi-3-mini
+            for message in messages:
+                formatted_prompt += f"<|{message['role']}|>{message['content']}<|end|>"
+            formatted_prompt += "<|assistant|>"
+
+        elif "Phi-4" in self.model_name:
```

Comment:
```suggestion
        elif "phi-4" in self.model_name:
```
Adding a note here in case you missed this one @HuanzhiMao